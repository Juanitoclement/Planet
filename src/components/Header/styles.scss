.recent-search {
  position: absolute;
  background-color: white;
  color: black;
  top: 110%;
  left: 50%;
  transform: translate(-50%, -7px);
  width: 245px;
  border-radius: 4px;
  box-shadow: 0 2px 8px 0 rgb(48 49 53 / 16%);
  overflow: scroll;
  max-height: 180px;

  .item {
    padding: 10px;

    span {
      padding-left: 10px;
      vertical-align: middle;
    }
  }

  .item:hover {
    background-color: rgba(52, 52, 52, 0.2);
    border-radius: 2px;
  }
}
