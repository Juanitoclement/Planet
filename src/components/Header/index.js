import React from "react";
import _ from "lodash";
import { CssBaseline, Toolbar } from "@material-ui/core";
import AppBar from "@material-ui/core/AppBar";
import SearchBar from "material-ui-search-bar";
import HistoryIcon from "@material-ui/icons/History";
import HideOnScroll from "../HideOnScroll";

import "./styles.scss";

const Header = (props) => {
  const [value, setValue] = React.useState("");
  const [hideRecent, setHideRecent] = React.useState(false);
  const localData = JSON.parse(window.localStorage.getItem("history"));
  const handleChange = (val) => {
    setValue(val);
    setHideRecent(false);
  };

  /* istanbul ignore next */
  return (
    <React.Fragment>
      <CssBaseline />
      <HideOnScroll {...props}>
        <AppBar>
          <Toolbar>
            <SearchBar
              value={value}
              onChange={(newValue) => handleChange(newValue)}
              onCancelSearch={() => setValue("")}
              onRequestSearch={() => {
                if (_.isEmpty(localData) && value !== "") {
                  window.localStorage.setItem(
                    "history",
                    JSON.stringify([value])
                  );
                }

                if (localData?.length > 0) {
                  let shouldAdd = localData.includes(value);
                  if (!shouldAdd) {
                    window.localStorage.setItem(
                      "history",
                      JSON.stringify([value, ...localData])
                    );
                  }
                }
                props.search(value);
                setHideRecent(true);
              }}
              style={{ height: 40, margin: "0 auto", maxWidth: 1000 }}
            />
            {localData?.length > 0 && value.length > 0 && !hideRecent && (
              <div className="recent-search">
                {localData.map((res, i) => {
                  return (
                    <div className="item" key={i} onClick={() => setValue(res)}>
                      <HistoryIcon />
                      <span>{res}</span>
                    </div>
                  );
                })}
              </div>
            )}
          </Toolbar>
        </AppBar>
      </HideOnScroll>
      <Toolbar />
    </React.Fragment>
  );
};

export default Header;
