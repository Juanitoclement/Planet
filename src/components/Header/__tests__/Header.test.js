import { fireEvent, render } from "@testing-library/react";
import Header from "../../Header";

describe("Header", () => {
  test("Should Render Header", () => {
    const { container, getByPlaceholderText } = render(<Header />);

    const inputSearch = getByPlaceholderText("Search");

    fireEvent.click(inputSearch);

    fireEvent.change(inputSearch, {
      target: { value: "planet" },
    });

    expect(container).toMatchSnapshot();
  });
});
