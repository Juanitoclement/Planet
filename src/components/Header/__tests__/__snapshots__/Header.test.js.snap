// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should Render Header 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiPaper-root ForwardRef-root-1 MuiPaper-elevation1 MuiPaper-rounded"
        style="height: 40px; margin: 0px auto; max-width: 1000px;"
      >
        <div
          class="ForwardRef-searchContainer-7"
        >
          <div
            class="MuiInputBase-root MuiInput-root ForwardRef-input-6 MuiInputBase-fullWidth MuiInput-fullWidth"
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              placeholder="Search"
              type="text"
              value="planet"
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-searchIconButton-4 ForwardRef-iconButtonHidden-3"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root ForwardRef-icon-5"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root ForwardRef-icon-5"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <div
    class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  />
</div>
`;
